org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/xsd/participants/XSDRenameParticipant.java

```diff
@@ -62,8 +66,8 @@ public Either<Range, PrepareRenameResult> prepareRename(IPrepareRenameRequest re
 	// --------------- Rename
 
 	@Override
-	public void doRename(IRenameRequest request, List<TextEdit> edits, CancelChecker cancelChecker) {
-		edits.addAll(getRenameTextEdits(request, cancelChecker));
+	public void doRename(IRenameRequest request, List<Either<TextDocumentEdit, ResourceOperation>> documentChanges, CancelChecker cancelChecker) {
+		mergeOrAddTextEdits(request.getXMLDocument(), getRenameTextEdits(request, cancelChecker), documentChanges);
```

Comment:
According the suggestion of the new API IRenameResponse https://github.com/eclipse/lemminx/pull/1522/files#r1213959272 code should look like  this:

`response.setTextDocumentEdits(getRenameTextEdits(request, cancelChecker));`

Reply:
@angelozerr Then `getRenameTextEdits()` should probably return `List<TextDocumentEdit>` instead of `List<TextEdit>`, like:
```
private List<TextDocumentEdit> getRenameTextEdits(IRenameRequest request, CancelChecker cancelChecker) 
```
Otherwise, `setTextDocumentEdits()`  still needs  `DOMDocument document` argument to be added in order to convert `TextEdit`-s  to `TextDocumentEdit`-s and merge them into the response.

Reply:
Yes I think